# BIMM-143-Project-1D-Final

Files needed for analysis

Scientific Question By comparing the protein sequences, and the proposed 3D structures, of MAO A, MAO B and zebra fish monoamine oxidases structural can functional differences between them be determined?

Hypothesis If the functional differences of MAO A, MAO B and the MAO of the zebra fish is caused by the nucleotide sequence and protein structures, then these differences will be observable in the binding and activation pockets of these enzymes, by aligning the nucleotide sequences and possible 3D structures.

All of the faa files are taken from the NCBI database, and downloaded as amino acid sequences

MAO_A_protein.faa: This is protein coded by the human genome, the file itself includes two different isotypes for this protein
https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=4128

MAO_B_protein.faa: Like the MAO A protein above it is a protein coded by the human genomes and the file includes two seperate isotypes
https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=4129

zfish_protein.faa: Unlike the two above this is not from the human genome, but instead that of the zebra fish (Danio rerio) which only has one monoamine oxidase.
https://www.ncbi.nlm.nih.gov/gene/404730

1gos.pdb: pdb file used for homology modeling.
https://www.rcsb.org/structure/1gos

MAOA_1.pdb: Homology model of isotype 1 from MAO A protein

MAOB_1.pdb: Homology model of isotype 1 from MAO B protein

zfish.pdb:  Homology model of MAO from the zebra fish genome.

MAOA_1.png: Image for quality of model for MAOA_1.pdb

MAOB_1.png: Image for quality of model for MAOB_1.pdb

zfish.png:  Image for quality of model for zfish.pdb

Project 2C Final.rmd: The original R notebook file, with the script and graph outputs, this is VERY rough draft that I plan on improving.

Project 2C Final.nb.html: The knitted html version of the same file.
